articles/142_idl.md

```diff
@@ -35,3 +35,95 @@ Further it describes how the interfaces are used to generate code in C, C++ and
+| bool        | \_Bool      | bool                           | bool                 |
+| octet       | uint8\_t    | std::byte[<sup>1</sup>](#byte) | bytes with length 1  |
+| int8        | int8\_t     | int8\_t                        | int                  |
+| uint8       | uint8\_t    | uint8\_t                       | int                  |
+| int16       | int16\_t    | int16\_t                       | int                  |
+| uint16      | uint16\_t   | uint16\_t                      | int                  |
+| int32       | int32\_t    | int32\_t                       | int                  |
+| uint32      | uint32\_t   | uint32\_t                      | int                  |
+| int64       | int64\_t    | int64\_t                       | int                  |
+| uint64      | uint64\_t   | uint64\_t                      | int                  |
+
+1. <a name="byte"></a>If std::byte is not available then uint8\_t is used instead.
+
+The next table defines how IDL templated and constructed types are mapped to the programming languages (unless specified otherwise below).
+When specified `T` is either one of the types above or an IDL struct.
+`N` is the upper limit of a bounded type.
+
+| IDL type       | C type                            | C++ type          | Python type |
+| -------------- | --------------------------------- | ----------------- | ----------- |
+| T[N]           | T[N]                              | std::array<T, N>  | list        |
+| sequence<T>    | struct {size\_t, T * }            | std::vector<T>    | list        |
+| sequence<T, N> | struct {size\_t, T * }, size\_t N | std::vector<T, N> | list        |
+| string         | char *                            | std::string       | str         |
+| string<N>      | char *                            | std::string       | str         |
+| wstring        | char16\_t *                       | std::u16string    | str         |
+| wstring<N>     | char16\_t *                       | std::u16string    | str         |
+
+These array and sequence types have special mappings.
+If a cell is blank then the default mapping is used.
+
+| IDL type                     | C type                            | C++ type       | Python type                      |
+| ---------------------------- | --------------------------------- | -------------- | -------------------------------- |
+| octet[N]                     |                                   |                | bytes                            |
+| sequence<octet>              |                                   |                | bytes                            |
+| sequence<octet, N>           |                                   |                | bytes                            |
+| array of string              | struct {size\_t, T * }, size\_t N |                |                                  |
+| int8[N]                      |                                   |                | numpy.array(dtype=numpy.int8)    |
```

Comment:
Should the fixed-size arrays also have `shape=(N, )`?

Reply:
The following lines work for me:

```
import numpy
numpy.ndarray(shape=(5, ), dtype=numpy.uint32)
```

Reply:
Ah, I was trying `numpy.array(...)`. It looks like the underlying type is `ndarray` and `array` is just a helper to construct one. I'll change the types.

Reply:
479e3a1f8488fbbdb6b8ebc00193e3dd809b9513